File: packages/schema/src/scene.ts
import { z } from "zod";

export const sceneSchema = z.object({
  id: z.string().min(1),
  type: z.enum(["narration", "dialogue", "silent"]),
  image: z.string().min(1),
  subtitleText: z.string().nullish(),
  voiceId: z.string().min(1).nullish(),
  audio: z.string().nullish(),
  durationMs: z.number().int().positive(),
  speaker: z.string().min(1).nullish(),
  stylePreset: z.enum(["default", "fast", "dramatic", "calm"]),
  cameraMotion: z.enum(["none", "pan", "zoom-in", "zoom-out"]).nullish(),
  transition: z.enum(["cut", "fade", "slide"]).nullish(),
});

export type Scene = z.infer<typeof sceneSchema>;

File: apps/api/app/models/scene.py
from pydantic import BaseModel, ConfigDict, Field


class Scene(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    id: str
    type: str
    image: str
    subtitle_text: str | None = Field(default=None, alias="subtitleText")
    voice_id: str | None = Field(default=None, alias="voiceId")
    audio: str | None = None
    duration_ms: int = Field(alias="durationMs")
    speaker: str | None = None
    style_preset: str = Field(alias="stylePreset")
    camera_motion: str | None = Field(default=None, alias="cameraMotion")
    transition: str | None = None
